endpoint/OBDComp/monitor/monitor.c

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (C) 2019-2020 BiiLabs Co., Ltd. and Contributors
+ * All Rights Reserved.
+ * This is free software; you can redistribute it and/or modify it under the
+ * terms of the MIT license. A copy of the license can be found in the file
+ * "LICENSE" at the root of this distribution.
+ */
+
+#include "interfaces.h"
+#include "legato.h"
+
+#include "brotli/decode.h"
+#include "brotli/encode.h"
+#include "build/obd_generated.h"
+#include "endpoint/OBDComp/can-bus/can-utils.h"
+#include "endpoint/OBDComp/obd_pid.h"
+#include "hal/device.h"
+
+#include "legato.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+static int string_compress(uint8_t *origin, size_t len, uint8_t **out_buffer, size_t *out_size) {
+  int lgwin = BROTLI_DEFAULT_WINDOW;
+  int bufferSize = len;
+  uint8_t *buffer = malloc(bufferSize);
+  uint8_t *inBuffer = malloc(bufferSize);
+
+  memcpy(inBuffer, origin, len);
+
+  int quality = 11;
```

Comment:
Explain why you picked up `11`.

Reply:
1. arr is an uint8_t `array`, not string.
2. We only need to send the last 4 bytes of data. The name of the function may be misleading, I will change it.